### DISCUSSION TITLE
Cookie differences

### DISCUSSION THREAD

**Discussion by EdLeafe** [encode/httpx#2253] [Potential Issue]
I have a project that used `requests` when it was started, and then later was moved to async and `aiohttp`. Now I'm trying to replace `aiohttp` with `httpx`, but I'm getting a failure that seems to be tied to how `httpx` handles cookies.
The project is a tool for handling Mailman admin issues; it passes authentication and gets a list of outstanding issues as well as a cookie, which needs to be sent in the response for handling the outstanding issues. With `requests` and `aiohttp`, parsing the cookie values returns:

```
KEY testing+admin
VAL Set-Cookie: testing+admin=280200000069e<snip>; Domain=mail.leafe.com; Path=/mailman/; Secure; Version=1
```

But with `httpx`, I see this:

```
KEY testing+admin
VAL 2802000000695b6<snip>
```

When I return that cookie to the server, the update request is rejected as unauthenticated, which doesn't happen with either `requests` or `aiohttp`.

I've been replacing all my earlier async code that used `aiohttp` with `httpx`, because I strongly prefer `httpx`. But this is a blocker for this project. The code flow is:

```
get_response = await session.get(url, headers=headers)
...
<process>
...
post_response = await session.post(url, headers=headers, cookies=get_response.cookies)
```

What can be done differently in the handling of the cookies?

**Comment by EdLeafe** [comment]
<bump>

Anyone??

**Comment by EdLeafe** [comment]
Wow, over 2 months with no reply. Sad.

**Reply by lovelydinosaur** [reply]
Hi Ed. I took a look at this when it came in, but the issue didn't help make it clear enough how to replicate what you're seeing, and so it was/is difficult to formulate a helpful response, or to see if it was/is worth investing my time into it.

Perhaps you could provide a replicable example demonstrating the differences?

**Reply by lovelydinosaur** [reply]
Specifically...

> With requests and aiohttp, parsing the cookie values returns ... But with httpx, I see this

This leaves some very big gaps for a maintainer or contributor to try to fill in.

**Reply by EdLeafe** [reply]
> Hi Ed. I took a look at this when it came in, but the issue didn't help make it clear enough how to replicate what you're seeing, and so it was/is difficult to formulate a helpful response, or to see if it was/is worth investing my time into it.
> 
> Perhaps you could provide a replicable example demonstrating the differences?

That would have been a helpful response back then. I wasn't sure what information would be helpful, so I framed the question as generally as possible, hoping someone more familiar with the code would have seen something like this before. 

As it turns out, I was on `0.22.0`, and while creating a small program to repro the issue, noticed that the current version in `0.23.0`. I upgraded to that, and the issue disappeared. I don't see anything in the `0.23.0` release notes that indicate that there were any changes with cookies, but it seems to have corrected the problem.

**Comment by EdLeafe** [comment]
So please feel free to close this issue.
